### DISCUSSION TITLE
Appearance vs Themes (What's in a name?)

### DISCUSSION THREAD

**Discussion by Raytle** [hluk/CopyQ#1707] [Ideas]
Admittedly this is a quibble.

 Why is [this section in ReadTheDocs](https://copyq.readthedocs.io/en/latest/theme.html) named "Themes"? I am familiar with terms such as styles, themes, and skins. Nonetheless, to my mind at least, "Appearance" would be more clear to the average user than "Themes" because themes are often used in quotidian English, er, uh, I mean common parlance, or, well, how about this: English people ordinarily use, to describe abstract concepts, such as the theme of a book, movie, or play. By contrast, appearance is normally used much more narrowly: to describe concretely what something looks like.

Feel free to ignore my arguments. But I doubt you'll be able to defeat the argument implicit in the distinction between...

https://www.etymonline.com/word/appear 
vs 
https://www.etymonline.com/word/theme

Put simply: whenever it is feasible, documentation should not make users struggle to grasp what is meant. In other words: keep stuff simple for them whenever it is feasible.

**Comment by hluk** [comment]
Not sure if it is worth changing. I'm used seeing Themes in apps.

**Comment by Raytle** [comment]
That's a lame argument. Your opinion is based on what you are comfortable with. Who cares what you are comfortable with? What matters is clearly communicating with users.

**Reply by hluk** [reply]
I'm saying it is not worth changing because nobody complained about it until now and changing this would probably change also the URL breaking older links.

I'm not native english speaker, but I do not see much difference between the names. I see, for example Firefox using "Themes".

**Reply by Raytle** [reply]
> I'm saying it is not worth changing because nobody complained about it until now

That's a very weak argument. 

It's tending towards nonsense such as, "No one who died in a car crash ever complained **after** they were dead that cars were unsafe."

Ordinary users almost invariably simply don't bother dealing with stuff they don't understand. Have you ever complained that, for example, air conditioners have filters that are too difficult to clean? No? Oh. Ok. Then I will conclude that air conditioners have filters that are not too difficult to clean. That's bad logic.


> and changing this would probably change also the URL breaking older links.


Why can't you use some sort of find/replace command to change all instances of "theme_foo" to "appearance_foo"?


> I'm not native english speaker, but I do not see much difference between the names. I see, for example Firefox using "Themes".

Your argument has merit but not enough. Worse still, it's an inadvertently pernicious argument which engineers almost invariably trot out to defend bad user interface design and/or bad user experience design. That sort of argument huge part of the reason that Linux has around merely around 2% of the desktop market for desktop and laptop computers.

Let me explain. The engineers (like you) who build stuff—not merely software but things like bridges and airplanes—often use the wrong words. By wrong I mean they misuse an English word thereby creating their own lexicon (lingo).

As a result, other engineers learn these words yet—of course—most ordinary users don't. It is particularly problematic when words are "slightly misused" because although, for example a jacket and a coat are similar yet—of course—they are actually distinct. Many Americans, for example, do not understand the difference between theft and robbery. Obviously this leads to miscommunication.

If your goal is to communicate exclusively with engineers then yes, of course, "Theme" is the best word.

If your goal is to communicate with ordinary users then "Appearance" is the best word.

Here's the thing: engineers will almost certainly understand the word "Appearance" even though it might seem a little strange to them. But many ordinary users simply will not almost certainly not understand what "Theme" means whereas they will almost certainly understand what the word "Appearance" means.

Here's what happens: engineers end up talking to each other so much that they become accustomed to "computer lingo" which makes perfect sense to them, but is often not easily understood by ordinary people.

I urge you to change "Theme" to "Appearance". Yes, it's a very small detail. But greatness almost invariably consists of myriad small details done correctly.

Bottom Line: If your goal is to be clear to ordinary users then ordinary English is best.

**Reply by gbcox** [reply]
@Raytle.... you said at the very beginning that this was a quibble - and you were correct - why then go on and write a tome about it.  I'm an native english speaker and in the present context the terms are basically interchangable.  No need to tilt at windmills.

**Reply by Raytle** [reply]
> you said at the very beginning that this was a quibble - and you were correct - why then go on and write a tome about it.

As I indicated in my penultimate comment...

>>  But greatness almost invariably consists of myriad small details done correctly.

In other words, a drop of water alone is trivial, but an ocean is comprised of myriad drops of water. 


> I'm an native english speaker and in the present context the terms are basically interchangable. 

You are clearly not an ordinary user. Ordinary users don't comment on Github issues like this. I'm thinking of people who, well, you know... who don't actually enjoy using computers but use them because they need to.

> No need to tilt at windmills.

As I implicitly argued above, that's a bad metaphor. Worse still it's also a clear and obvious insult. How would you like it if someone were to impugn your character by unfairly comparing you to a well-known buffoon: a self-centered, delusional, arrogant, nit-wit  who exemplified the kind of man who Cervantes believed helped to cause the fall of the once mighty Spanish Empire?

If you have something valuable to add to this discussion, then I welcome your comments.

**Reply by gbcox** [reply]
I commented because I actively monitor this forum and you were needlessly arguing with the maintainer.  He has better things to do, as should you.  As far as "tilting at windmills" is concerned, it's an idiom.  Look it up for the meaning.  Construing it as a "clear and obvious insult" speaks volumes.

**Reply by Raytle** [reply]
> I'm saying it is not worth changing because nobody complained about it until now 

In the comments section of this YouTube video... [CopyQ: A Clipboard Manager That Does Everything??](https://www.youtube.com/watch?v=DhfwWH5N3IA) I read the following comment...

Coming to linux from something like macOS, the general off-putting trend seems to be that linux app developers fail to understand proper UI designing methodologies... this application feels like a perfect example as it’s incredible in the feature set that it provides along with all the flexibility however the topic reminded me of an amazing macOS app known as Paste2. It is a clipboard manager and has one of the most beautiful interfaces I’ve seen in such an application.

I don’t know if anyone else feels this way. Maybe good UI/UX design isn’t a priority with most linux applications. I suppose that might be because the community seems to prefer CLI over a GUI, however that preference doesn’t make sense to me for something like a clipboard manager.

**Comment by Raytle** [comment]
Stop trolling.

**Reply by gbcox** [reply]
ROFL... classic projection.

**Comment by akoppa** [comment]
Really quibble. I'm dealing with computers for more than 20 years and never heard something like that. Theme is computer jargon and as such the whole world understand what a theme is. 
Quoted: "In computing, a theme is a preset package containing graphical appearance and functionality details. A theme usually comprises a set of shapes and colors for the graphical control elements, the window decoration and the window. Themes are used to customize the look and feel of a piece of computer software or of an operating system."

**Comment by Gitoffthelawn** [comment]
I didn't waste my time reading much of this (when people choose not to behave in a respectful manner, I stop reading their content).

I'm not saying I agree or disagree with the proposed change, but the lack of respect towards our generous primary developer is ugly and unacceptable.

This is an open-source project... if someone really, really, really wants to change a word, they can fork it, spend ***their*** time changing the word, and be done with it already.
